roadmap/implementers-guide/src/runtime/disputes.md

```diff
@@ -17,90 +19,351 @@ We account for these requirements by having the validity module handle two kinds
+Votes in favour of block validity always includes the primary votes from the `Backing` stage.
 
-Local disputes are only allowed on parablocks that have been included on the local chain and are in the acceptance period.
+Votes aginst the block validity always include the initial challenger from the `Backing` stage iff any.
 
-For each such parablock, it is guaranteed by the inclusion pipeline that the parablock is available and the relevant validation code is available.
+Session change implies a change of session keys. Hence at least two validator sets must be kept - the set of validators in the current session and the ones in the previous session.
+The timeouts must be set accordingly to guarantee that two sessions are sufficient, and timeout actions are not triggered in a 3rd session.
+An era change implies a session change, since the nominators change.
 
-Disputes may occur against blocks that have happened in the session prior to the current one, from the perspective of the chain. In this case, the prior validator set is responsible for handling the dispute and to do so with their keys from the last session. This means that validator duty actually extends 1 session beyond leaving the validator set.
+When a dispute has occurred on another fork, we need to transplant that dispute to every other fork. This poses some major challenges.
 
-...
+There are two types of remote disputes. **Concluded** and **Unconcluded**.
 
-After concluding with enough validtors voting, the dispute will remain open for some time in order to collect further evidence of misbehaving validators, and then issue a signal in the header-chain that this fork should be abandoned along with the hash of the last ancestor before inclusion, which the chain should be reverted to, along with information about the invalid block that should be used to blacklist it from being included.
+The first is a remote roll-up of a **concluded dispute**. These are simply all attestations for the block, those against it, and the result of all (secondary) approval checks. A concluded remote dispute can be resolved in a single transaction as it is an open-and-shut case of a quorum of validators disagreeing with another.
 
-## Remote Disputes
+The second type of remote dispute is the **unconcluded dispute**. An unconcluded remote dispute is started by any validator using a `UnconcludedDisputeGossipNotificationMessage`:
 
-When a dispute has occurred on another fork, we need to transplant that dispute to every other fork. This poses some major challenges.
+When beginning a remote dispute, at least one escalation by a validator is required, but this validator may be malicious and desires to be slashed. There is no guarantee that the para is registered on this fork of the relay chain or that the para was considered available on any fork of the relay chain.
```

Comment:
If helpful, you could require a dispute message forward one original message with which it disagrees, so then remote or local makes little difference.  All the existing votes remote or local could then be replayed onto the current fork.

Reply:
But that would imply I can only escalate once I have two opposing votes - the negative by the "local" validator (sh*t names, sorry) / secondary checker, selected by VRF,  and the ones backing the candidate.

Reply:
I answered the first part well enough below in https://github.com/paritytech/polkadot/pull/1705#discussion_r521788303 yes?

I think replay means our chain `C` knows some `disputed: CandidateHash` so any vote about `disputed` valid on any chain with any validator set can be posted on `C` inside some `ReplayDisputedVote` wrapper message. 

We've one tricky corner however:  We'll map these votes into the slashing spans on `C`, which means we need to map them to sessions somehow.  So how can we extract a session from `disputed` when it does not come from `C`?  Should we include the session inside the authority identifier somehow?